src/combine.ts

```diff
@@ -1,12 +1,12 @@
-import { Source, ExtractContent, Talkback } from './types';
+import { Producer, ExtractContent, Talkback, Callbag } from './types';
 
-export function combineWith<T extends [...Source<any>[]], U>(
+export function combineWith<T extends [...Producer<unknown>[]], U>(
   f: (...args: ExtractContent<T>) => U,
   ...sources: T
-): Source<U> {
+): Producer<U> {
   return (_, sink) => {
     const n = sources.length;
-    let combined: any[] = Array(n).fill(undefined);
+    let combined: unknown[] = Array(n).fill(undefined);
```

Comment:
Interesting use of `unknown`! Indeed, if we don't directly access the contents of `combined[i]`, then this is `unknown`. Nice.